### ISSUE TITLE
[doc] Add dependency graphs to documentation

### ISSUE THREAD

**Issue by aPonza** [PointCloudLibrary/pcl#4047]
**Is your feature request related to a problem? Please describe.**

I'm always frustrated when I can't see updated dependency graphs in the documentation, showing the actual dependencies between modules as parsed by Doxygen. (yeah right)

**Context**

Provide an updated view on dependencies not requiring delving into CMakeLists.txts or [using old images available online](https://www.researchgate.net/figure/Point-Cloud-Library-PCL-dependency-graph_fig20_284142470).

**Expected behavior**

The documentation available at https://pointcloudlibrary.github.io/documentation/modules.html could include a generated graph from the code.

**Current Behavior**

There is no graph available in the modules inside that webpage.

**Describe the solution you'd like**

There is a graph available for each module inside that webpage. Generated via Doxygen to make it "easy" as it should be available thanks to the [GROUP_GRAPHS](http://www.doxygen.nl/manual/config.html#cfg_group_graphs) configuration option which is currently enabled as well in our doxyfile.

**Describe alternatives you've considered**

It's a nice-to-have, not anything mandatory, so no other alternatives yet.

**Additional context**

This issue template is really in depth.

**Comment by SergioRAgostinho** [comment]
The one in our [website](https://pointcloudlibrary.github.io/about/) could also use an (automatic?) update.

**Comment by aPonza** [comment]
Yep yep, that'd be the point as well, as the `[common]` graph should be that one.

**Comment by aPonza** [comment]
Found [this](https://github.com/PointCloudLibrary/pcl/wiki/FAQ_devel#current) regarding the old dependency graph
